### PR TITLE
#581 | Address page nft transfers tabs

### LINKED ISSUE #581: Address Page NFT Transfers Tab
Add NFT Transfers Tab if address has NFT transfers:
 - Title: Latest NFT Transfer Events
 - Columns (As from Transactions table):
     -?, Transaction Info, MEthod, BLock, Age, From, Direction Label (IN (green) or OUT(Red)), Type (ERC-1155 / ERC-721), Item (Logo, Collection Name (Link to Address Page of NFT or NFT page if already implemented) , Contract Name)

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #581 

[Link](https://deploy-preview-649--teloscan-redesign-mainnet.netlify.app/address/0x61cCeF8277A6Ec2984af9d75366ed14F52f4B7F6?tab=erc721txns)

## Description
Instead of creating a new component for the new NFT tab, this PR adds two separate tabs for erc-712 and erc-1155 transfers to reuse a pre-existing component that was adjusted to avoid using the old ContractManager and take the data directly from the API response, which is significantly faster.

## Test scenarios
- Go and visit [this address](http://localhost:8080/address/0x61cCeF8277A6Ec2984af9d75366ed14F52f4B7F6?tab=erc721txns)
- Explore the ERC transfer tabs

## Screenshots
[telosca--method-lazy-loading.webm](https://github.com/telosnetwork/teloscan/assets/4420760/c461e510-6e12-4b5a-ab31-1d1a532e6f14)

image

image
image

## Checklist:
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness
-   [x] I have added appropriate test coverage
-   [x] I have updated relevant documentation and/or opened a separate issue to cover the updates (Issue URL: )

### CODE REVIEW

File: src/components/TransferTable.vue [hunk 1]
```diff
@@ -113,10 +114,22 @@ export default {
         });
     },
     methods: {
+        truncatedId(id) {
+            // only truncate if the id is longer than 7 characters
+            // we need to show the first 4 and last 3 characters
+            if (id.length > 7) {
+                return id.substr(0, 4) + '...' + id.substr(id.length - 3);
```
Comment (donnyquixotic) [inline src/components/TransferTable.vue:121]:
```suggestion
                return `${id.substr(0, 4)}...${id.substr(id.length - 3)}`;
```

File: src/components/NftTransfersTable.vue [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+import { onMounted, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+import { indexerApi } from 'src/boot/telosApi';
+
+import TransactionField from 'components/TransactionField';
+import AddressField from 'components/AddressField';
+import DateField from 'components/DateField.vue';
+import { formatWei, toChecksumAddress } from 'src/lib/utils';
+import { Pagination } from 'src/types';
```
Comment (donnyquixotic) [inline src/components/NftTransfersTable.vue:11]:
```suggestion
import { indexerApi } from 'src/boot/telosApi';
import { formatWei, toChecksumAddress } from 'src/lib/utils';
import { Pagination } from 'src/types';

import TransactionField from 'components/TransactionField';
import AddressField from 'components/AddressField';
import DateField from 'components/DateField.vue';
```

File: src/components/NftTransfersTable.vue [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+        const contractData = response.data.contracts[transfer.contract];
+        let valueDisplay;
+        if(props.tokenType === 'erc20'){
+            if (contractData && contractData.decimals) {
+                valueDisplay = formatWei(transfer.amount, contractData.decimals);
+            } else {
+                valueDisplay = $t('components.unknown_precision');
+            }
+        } else if (props.tokenType === 'erc721') {
+            valueDisplay = '#' + transfer.id.toString();
```
Comment (donnyquixotic) [inline src/components/NftTransfersTable.vue:156]:
```suggestion
            valueDisplay = `#${transfer.id.toString()}`;
```

File: src/components/NftTransfersTable.vue [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+        } else if (props.tokenType === 'erc1155') {
+            valueDisplay = transfer.amount.toString();
+        } else if(transfer.id) {
+            valueDisplay = '#' + transfer.id.toString();
+        }
```
Comment (donnyquixotic) [inline src/components/NftTransfersTable.vue:161]:
```suggestion
        } if (props.tokenType === 'erc1155') {
            valueDisplay = transfer.amount.toString();
        } else if(transfer.id) {
            valueDisplay = `#${transfer.id.toString()}`;
        }
```

**Review by donnyquixotic — CHANGES_REQUESTED** [review]
see minor feedback

File: src/components/TransferTable.vue (hunk 1 above)
Comment (Viterbo) [reply src/components/TransferTable.vue]: Done!

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**